index.js

```diff
@@ -388,7 +388,15 @@ module.exports = class YarnWorkspacesPlugin extends Plugin {
   }
 
   getRegistry() {
-    return this.getContext('publishConfig.registry') || NPM_DEFAULT_REGISTRY;
+    const { publishConfig } = this.getContext();
+    const registries = publishConfig
+      ? publishConfig.registry
+        ? [publishConfig.registry]
+        : Object.keys(publishConfig)
+            .filter((key) => key.endsWith('registry'))
```

Comment:
So just to be clear, we're expecting _some_ string to end with `registry` here. Should we be more precise in what we're matching on here, such as regexing the format we expect? eg. `@[^:]*:registry`?